src/api_keys.rs

```diff
@@ -0,0 +1,192 @@
+use std::fmt;
+use std::sync::Arc;
+
+use reqwest::Method;
+
+use crate::types::{CreateApiKeyRequest, CreateApiKeyResponse, ListApiKeysResponse};
+use crate::{Config, Result};
+
+/// `Resend` APIs for `METHOD /api-keys` endpoints.
+#[derive(Clone)]
+pub struct ApiKeys(pub(crate) Arc<Config>);
+
+impl ApiKeys {
+    /// Add a new API key to authenticate communications with Resend.
+    ///
+    /// <https://resend.com/docs/api-reference/api-keys/create-api-key>
+    #[cfg(not(feature = "blocking"))]
+    #[cfg_attr(docsrs, doc(cfg(not(feature = "blocking"))))]
+    pub async fn create(&self, api_key: CreateApiKeyRequest) -> Result<CreateApiKeyResponse> {
+        let request = self.0.build(Method::POST, "/api-keys");
+        let response = request.json(&api_key).send().await?;
+        let content = response.json::<CreateApiKeyResponse>().await?;
```

Comment:
The issue with this is that it tries to parse the API response into the `CreateApiKeyResponse` struct but when the server returns an error, it tries to parse that instead and thus the API error is both not reported and you get a confusing parsing error instead. 

You could test this by for example sending an email to an address that is not your resend address if you are still in testing as their docs state "You can only send testing emails to your own email address".

I think the way to go about this would be to represent all [resend errors](https://resend.com/docs/api-reference/errors). They did a great job at writing them down in that page, I think we could show the same information in an error message (or direct the users to the page itself by giving them a link such as `https://resend.com/docs/api-reference/errors#invalid-from-address`).

We are already returning a Result from all of these methods so it would not be too hard to add a `ResendError` entry to the enum and have a "is status 200" before parsing it here so we can return the errors properly. The same goes for all API calls.

Reply:
Oops, forgot about `.error_for_status()?`.
Yeah, I thought about getting rid of `reqwest::Error` and using actual error types.
Let me try to figure something out.

Reply:
Updated `crate::Error` and `emails.rs::send()`. Should I make any changes before updating the rest?

Reply:
It's great now!

Reply:
Updated again to use the `ErrorKind`.